lib/src/http.dart

```diff
@@ -382,13 +383,10 @@ class _ThrottleClient extends http.BaseClient {
       rethrow;
     }
 
-    var stream = response.stream.transform(
-        StreamTransformer<List<int>, List<int>>.fromHandlers(
-            handleDone: (sink) {
-      resource.release();
-      sink.close();
-    }));
-    return http.StreamedResponse(stream, response.statusCode,
+    final responseController = StreamController<List<int>>(sync: true);
+    unawaited(response.stream.pipe(responseController));
+    unawaited(responseController.done.then((_) => resource.release()));
```

Comment:
Consider using `.whenComplete(resource.release)`. It works even if the `done` event contains an error.